eval/api_runner.py

```diff
@@ -0,0 +1,132 @@
+from concurrent.futures import ThreadPoolExecutor, as_completed
+from typing import Optional
+from eval.eval import compare_query_results
+import pandas as pd
+from utils.pruning import prune_metadata_str
+from utils.questions import prepare_questions_df
+from tqdm import tqdm
+from time import time
+import requests
+
+
+def generate_prompt(prompt_file, question, db_name, public_data):
+    with open(prompt_file, "r") as f:
+        prompt = f.read()
+
+    pruned_metadata_str = prune_metadata_str(question, db_name, public_data)
+    prompt = prompt.format(
+        user_question=question, table_metadata_string=pruned_metadata_str
+    )
+    return prompt
+
+
+def process_row(row, api_url, num_beams, public_data):
+    start_time = time()
+    # we set return_full_text to False so that we don't get the prompt text in the generated text
+    # this simplifies our postprocessing to deal with just the truncation of the end of the query
+    r = requests.post(
+        api_url,
+        json={
+            "prompt": row["prompt"],
+            "n": 1,
+            "use_beam_search": True,
+            "best_of": num_beams,
+            "temperature": 0,
+            "stop": [";", "```"],
+            "max_tokens": 600,
```

Comment:
nit: are we able to set `return_full_text` in the request params here? if not we can remove the comment above.

Reply:
Ah we're not, removed!